engine/time/src/main/java/io/deephaven/time/calendar/BusinessDay.java

```diff
@@ -0,0 +1,224 @@
+        this(new BusinessPeriod[0]);
+    }
+
+    /**
+     * Business periods for the day.
+     *
+     * @return business periods for the day
+     */
+    public BusinessPeriod<T>[] periods() {
+        return openPeriods;
+    }
+
+    /**
+     * Start of the business day. Equivalent to the start of the first business period.
+     *
+     * @return start of the business day, or null for a holiday schedule
+     */
+    public T businessStart() {
+        return businessStart;
+    }
+
+    /**
+     * End of the business day. Equivalent to the end of the last business period.
+     *
+     * @return end of the business day, or null for a holiday schedule
+     */
+    public T businessEnd() {
+        return businessEnd;
+    }
+
+    /**
+     * Gets the length of the business day in nanoseconds. If the business day has multiple periods, only the time
+     * during the periods is counted.
+     *
+     * @return length of the day in nanoseconds
+     */
+    public long businessNanos() {
+        return businessNanos;
+    }
+
+    /**
+     * Amount of business time in nanoseconds that has elapsed on the given day by the specified time.
+     *
+     * @param time time
+     * @return business time in nanoseconds that has elapsed on the given day by the specified time
+     */
+    public long businessNanosElapsed(final T time) {
+        Require.neqNull(time, "time");
+        long elapsed = 0;
+
+        for (BusinessPeriod<T> businessPeriod : openPeriods) {
+            if (time.compareTo(businessPeriod.start()) < 0) {
+                return elapsed;
+            } else if (time.compareTo(businessPeriod.end()) > 0) {
+                elapsed += businessPeriod.nanos();
+            } else {
+                elapsed += businessPeriod.start().until(time, ChronoUnit.NANOS);
+                return elapsed;
+            }
+        }
+
+        return elapsed;
+    }
+
+    /**
+     * Amount of business time in nanoseconds that remains until the end of the day.
+     *
+     * @param time time
+     * @return business time in nanoseconds that remains until the end of the day.
+     */
+    public long businessNanosRemaining(final T time) {
+        Require.neqNull(time, "time");
+        return businessNanos - businessNanosElapsed(time);
+    }
+
+    /**
+     * Is this day a business day?
+     *
+     * @return true if it is a business day; false otherwise.
+     */
+    public boolean isBusinessDay() {
+        return businessNanos > 0;
+    }
+
+    /**
+     * Determines if the specified time is a business time for the day.
+     *
+     * @param time time.
+     * @return true if the time is a business time for the day; otherwise, false.
+     */
+    public boolean isBusinessTime(final T time) {
+        for (BusinessPeriod<T> p : openPeriods) {
+            if (p.contains(time)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I think this could use a short comment on how equality is being defined for this object.

Reply:
This is equality so that the object works in java collections.  The current code is more clear, but it is just checking for identical contents.  Very standard, so it is using the default javadoc via `@Override`.